### PR TITLE
MueLu: Aggregate quality estimates

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu

## Motivation
Rebase of PR #3457.

Let's see what the autotester thinks about it..

## Tests

Tests: RHEL6, complex, serialnode

### CODE REVIEW

File: packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_decl.hpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    void DeclareInput(Level &currentLevel) const;
+
+    //@}
+
+    //! @name Build methods.
+    //@{
+
+    //! Build aggregate quality esimates with this factory.
+    void Build(Level & currentLevel) const;
+    void BuildP(Level & currentLevel) const;
```
Comment (mayrmt) [inline packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_decl.hpp]: Out of curiosity: why two build routines, where `Build()` just wraps `BuildP()` without any further action?

Comment (lucbv) [reply packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_decl.hpp]: This is due to the inheritance from PFactory, however why was it designed like that, I do not know?

**Comment by csiefer2** [comment]
@cgcgcg Going to try to get this to the point we can merge.

**Comment by csiefer2** [comment]
MueLu now compiles with complex.  Now for Stokhos...

**Pull request merged**